src/Emulator/Visuals/ContextMenus.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useCallback } from 'react';
+import { Menu } from 'react-contexify';
+import { Button, Stack } from '@chakra-ui/core';
+
+import { DeviceInterface } from 'src/common/types';
+
+import { TopologyActions } from '../useTopology';
+import { useEmulator } from '../EmulatorProvider';
+
+interface Props {
+  devices: DeviceInterface[];
+}
+
+export const ContextMenus = ({ devices }: Props) => {
+  const { dispatch } = useEmulator();
+
+  const handleDeleteConnections = useCallback((from: string, to: string) => {
+    return () => {
+      dispatch({
+        type: TopologyActions.DELETE_CONNECTION,
+        payload: {
+          to,
+          from,
+        },
+      });
+    };
+  }, [dispatch]);
+
+  const handleDelete = useCallback((device: DeviceInterface) => {
+    let deviceTypeAction = device.name[0].toLocaleLowerCase() === 'r' ? TopologyActions.DELETE_ROUTER :
+      device.name[0].toLocaleLowerCase() === 's' ? TopologyActions.DELETE_SWITCH :
+        device.name[0].toLocaleLowerCase() === 'h' ? TopologyActions.DELETE_HOST :
```

Comment:
Does Javascript have enums? Is there a better way we can do this than the naming convention?

Reply:
They do. One thing that might actually be helpful is to create some helper functions that take in a device name and return the type maybe like
```ts
const isRouter = (name: string): bool => device.name[0].toLocalLowerCase === 'r';
```

and use those instead of these checks. Thoughts?

Reply:
I do something like this in python! I think this code is maintainable, but yes it will probably be cleaner that way.

Reply:
It turns out we have a `type` field on the device object. So I think that doing `device.type === 'router' etc...` works well. The `typeof(device.type) === DeviceType` which is a string type that consists of only `'router' | 'switch' | 'host'` and cannot be any other string.